### PR TITLE
trivial: use NUM_SGI_VIRQS

### PR DESCRIPTION
**Pull request by axel-h**
Avoid using plain magic numbers.

### CODE REVIEW

File: libsel4vm/src/arch/arm/vgic/vgic.c [hunk 1]
```diff
@@ -551,7 +551,7 @@ static int vgic_dist_disable_irq(struct vgic_dist_device *d, vm_vcpu_t *vcpu, in
 {
     /* STATE g) */
     struct gic_dist_map *gic_dist = vgic_priv_get_dist(d);
-    if (irq >= 16) {
```
Comment (kent-mcleod) [inline libsel4vm/src/arch/arm/vgic/vgic.c:554]: Are you able to add a comment here that this is because 0-15 cannot be disabled.

**Review by kent-mcleod — APPROVED** [review]

**Review by kent-mcleod — APPROVED** [review]

File: libsel4vm/src/arch/arm/vgic/vgic.c (hunk 1 above)
Comment (axel-h) [reply libsel4vm/src/arch/arm/vgic/vgic.c:554]: added a commit on top that adds a comment and also a debug log output

File: libsel4vm/src/arch/arm/vgic/vgic.c [hunk 2]
```diff
@@ -551,7 +551,11 @@ static int vgic_dist_disable_irq(struct vgic_dist_device *d, vm_vcpu_t *vcpu, in
 {
     /* STATE g) */
     struct gic_dist_map *gic_dist = vgic_priv_get_dist(d);
-    if (irq >= NUM_SGI_VIRQS) {
+
+    /* SGIs cannot be disabled, such a request is ignored. */
+    if (irq < NUM_SGI_VIRQS) {
+        DDIST("can't disable SGI %d (vCPU %d)\n", irq, vcpu->vcpu_id);
```
Comment (axel-h) [inline libsel4vm/src/arch/arm/vgic/vgic.c:557]: @kent-mcleod: should this become a `ZF_LOGW()` or event `ZF_LOGE()`

Comment (kent-mcleod) [reply libsel4vm/src/arch/arm/vgic/vgic.c:557]: I don't think so.  I recall that for GICv2:
> Whether SGIs are permanently enabled, or can be enabled and disabled by writes to the GICD_ISENABLERn and
> GICD_ICENABLERn, is IMPLEMENTATION DEFINED . 

And then for the particular GIC that is being emulated, it is ignored.  

Guest kernel's can still try and disable all IRQs as part of a startup routine.  seL4 does this in `gic_v2.c`.  So printing a warning would generate a lot of spurious warnings I expect.

**Pull request merged**